compiler/rustc_smir/src/stable_mir/ty.rs

```diff
@@ -34,7 +34,59 @@ pub struct Const {
 }
 
 type Ident = Opaque;
-pub(crate) type Region = Opaque;
+
+#[derive(Debug, Clone)]
+pub struct Region {
+    pub kind: RegionKind,
+}
+
+#[derive(Debug, Clone)]
+pub enum RegionKind {
+    ReEarlyBound(EarlyBoundRegion),
+    ReLateBound(DebruijnIndex, BoundRegion),
+    ReFree(FreeRegion),
+    ReStatic,
+    ReVar(RegionVid),
+    RePlaceholder(Placeholder<BoundRegion>),
+    ReErased,
+    ReError(ErrorGuaranteed),
```

Comment:
we do not forward errors to SMIR in Ty and Const, so we don't need to do it here either.